src/java.desktop/share/native/libfontmanager/freetypeScaler.c

```diff
@@ -46,6 +46,12 @@
 
 #include "fontscaler.h"
 
+#define CHECK_EXCEPTION(env)            \
+    if ((*(env))->ExceptionCheck(env)) { \
+        (*(env))->ExceptionDescribe(env);\
+        (*(env))->ExceptionClear(env);   \
+    }
```

Comment:
https://docs.oracle.com/en/java/javase/11/docs/specs/jni/functions.html#exceptiondescribe

"The pending exception is cleared as a side-effect of calling this function"

So you certainly don't need both of these and I would prefer that Describe only be used if really debugging where we think there's a REAL chance of an exception rather than just to keep JNI happy.

And the upcall that is likely (readBlock) itself will log any IOException (and catch it) in the event of an I/O error so I really think this is unlikely to be useful here.

Reply:
Yes, makes sense. I made `CHECK_EXCEPTION()` call either `Clear...` or `Describe...` based on the value of `debugFonts`. 
Please, take a look.